### PR TITLE
refactor(plugins) removing ngx.req.raw_header() for HTTP/2 compatibility

### PR DESCRIPTION
**Pull request by thefosk**
* Enables HTTP/2 compatibility by removing `ngx.req.raw_header()` invocation calls.

### CODE REVIEW

File: spec/03-plugins/09-galileo/ngx.lua [hunk 1]
```diff
@@ -13,11 +13,10 @@ return {
...
     raw_header = function ()
-      return [[
-GET /request/path HTTP/1.1\r
-Host: mockbin.com\r
-Accept: application/json\r
-\r\n\r\n]]
+      return "GET /request/path HTTP/1.1\r\n"..
+             "Host: mockbin.com\r\n"..
+             "Accept: application/json\r\n"..
+             "Accept: application/x-www-form-urlencoded\r\n\r\n\r\n"
```
Comment (thefosk) [inline spec/03-plugins/09-galileo/ngx.lua]: Required to avoid double escaping of `\r`.

File: kong/plugins/galileo/alf.lua [hunk 2]
```diff
@@ -80,6 +80,22 @@ local function hash_to_array(t)
...
+-- space that may be sorrounding values, other than that it's accurate)
+local function calculate_headers_size(request_line, headers)
+  local size = 0
+  for k, v in pairs(headers) do
+    if type(v) == "table" then
+      for _, y in ipairs(v) do
+        size = size + #k + 2 + #tostring(y) + 2 --First 2 is semicolon + space
+      end
+    else
+      size = size + #k + 1 + 1 + #tostring(v) + 2 --First 2 is semicolon + space
```
Comment (shashiranjan84) [inline kong/plugins/galileo/alf.lua]: why 1 + 1 instead 2? what makes it different from line 90?

File: kong/plugins/galileo/alf.lua [hunk 3]
```diff
@@ -80,6 +80,22 @@ local function hash_to_array(t)
...
+    end
+  end
+  return #request_line + 2 + size + 4 -- 2 it's \r\n, 4 it's trailing \r\n\r\n
```
Comment (shashiranjan84) [inline kong/plugins/galileo/alf.lua]: I think you already counting one `\r\n` when you calculating the size of last header, so should be final sum be

`#request_line + 2 + size + 2`

**Comment by thefosk** [comment]
Updated.

**Comment by p0pr0ck5** [comment]
(fixed merge conflict and squashed)

**Pull request merged**